src/api/services/userService.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { NextFunction, Response } from 'express';
+import { Knex } from 'knex';
+
+import { TestConfigure } from '../../database/TestSelfServeDatabase';
+import {
+  createParticipant,
+  createResponseObject,
+  createUser,
+  createUserParticipantRequest,
+  getParticipantIdsOfUser,
+} from '../../testHelpers/apiTestHelpers';
+import { verifyAndEnrichUser } from '../middleware/usersMiddleware';
+import * as kcUsersService from './kcUsersService';
+import { UserService } from './userService';
+
+describe('User Service Tests', () => {
+  let knex: Knex;
+  let next: NextFunction;
+  let res: Response;
+
+  beforeEach(async () => {
+    knex = await TestConfigure();
+    next = jest.fn();
+    ({ res } = createResponseObject());
+    jest.spyOn(kcUsersService, 'removeApiParticipantMemberRole').mockResolvedValue();
```

Comment:
I think mocking keycloak here is fine - I don't think it's crucial enough to the production behaviour to warrant calling a "real" keycloak and checking for actual client roles.